### ISSUE TITLE
CA1508 false positive with is type pattern

### ISSUE THREAD

**Issue by lucastheisen** [dotnet/roslyn-analyzers#5011]
### Analyzer

**Diagnostic ID**: [CA2013](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2013): `Do not use ReferenceEquals with value types`
[CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.201](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug
False positive when using [`is` type pattern](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is#type-pattern):

```csharp
        private static IEnumerable<string> ToStringsForJoinArgs(object parameter)
        {
            IEnumerable<string> strings = null;
            if (parameter is string text)
            {
                strings = Enumerable.Repeat(text, string.IsNullOrWhiteSpace(text) ? 0 : 1);
            }
```

### Steps To Reproduce

I don't have a minimal reproduction but should be easy to verify by using the pasted code as an example.

### Expected behavior

No warning

### Actual behavior
```text
/mnt/c/Users/ltheisen/git/asiasetl-dotnet-cli-app/src/Mitre.CliApp.Sql.MySql/MySqlDbDumpWorkflowStrategy.cs(305,51): warning CA1508: 'string.IsNullOrWhiteSpace(text)' is always 'false'. Remove or refactor the condition(s) to avoid dead code. [/mnt/c/Users/ltheisen/git/asiasetl-dotnet-cli-app/src/Mitre.CliApp.Sql.MySql/Mitre.CliApp.Sql.MySql.csproj]
```

## Additional context

Using ubuntu 18 WSL image:

```console
ltheisen@MM233009-PC:~/git/asiasetl-dotnet-cli-app$ dotnet --version
5.0.201

ltheisen@MM233009-PC:~/git/asiasetl-dotnet-cli-app$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
```

**Comment by mavasani** [comment]
I believe this is fixed in the latest NuGet package of `Microsoft.CodeAnalysis.NetAnalyzers` version `5.0.3` and also in .NET 6 branch of the analyzers

**Comment by mavasani** [comment]
This is fixed now.
